src/hydraul.c

```diff
@@ -566,19 +568,17 @@ void  demands(Project *pr)
     for (i = 1; i <= net->Njuncs; i++)
     {
         sum = 0.0;
-		list_t *dlist = net->Node[i].D;
-		
-		if (dlist) {
-			for (list_node_t *lnode = first_list(dlist); done_list(lnode); lnode = next_list(lnode))
-			{
-				// pattern period (k) = (elapsed periods) modulus (periods per pattern)
-				j = get_pattern_index(lnode);
-				k = p % (long)net->Pattern[j].Length;
-				djunc = (get_base_demand(lnode)) * net->Pattern[j].F[k] * hyd->Dmult;
-				if (djunc > 0.0) hyd->Dsystem += djunc;
-				sum += djunc;
-			}
-		}
+		dlist = net->Node[i].D;
```

Comment:
Basically the same just renaming iterator access functions.

Reply:
But without exposing implementation details of a geneic list. My approach is much more robust (and has simpler syntax as well).